core/domain/suggestion_services.py

```diff
@@ -599,6 +624,193 @@ def get_translation_suggestions_waiting_longest_for_review_per_lang(
     ]
 
 
+def _get_plain_text_from_html_content_string(html_content_string):
+    """Retrieves the plain text from the given html content string. RTE element
+    occurrences in the html are replaced by its corresponding name capitalized
+    in square brackets.
+    eg: <p>Sample1 <oppia-noninteractive-math></oppia-noninteractive-math>
+        Sample2 </p> will give as output: Sample1 [MATH] Sample2.
+    Note: similar logic exists in the frontend in format-rte-preview.filter.ts.
+
+    Args:
+        html_content_string: str. The content html string to convert to plain
+            text.
+
+    Returns:
+        str. The plain text string from the given html content string.
+    """
+    # Replace frequently used HTML character entities.
+    html_content_string = re.sub('&nbsp;', ' ', html_content_string)
+    html_content_string = re.sub('&quot;', '"', html_content_string)
+    html_content_string = re.sub('&amp;', '&', html_content_string)
+
+    def _replace_rte_tag(rte_tag):
+        """Replaces all of the <oppia-noninteractive-**> tags.
+
+        Args:
+            rte_tag: MatchObject. The oppia-noninteractive rte tag.
+
+        Returns:
+            str. The string to replace the rte tag with.
+        """
+        # Convert the MatchObject to a string.
+        rte_tag_string = rte_tag.group(0)
+        # Get the name of the noninteractive tag (ex. math).
+        replace_string = rte_tag_string.split('-')[2].split(' ')[0]
+        if replace_string[-1] == '>':
+            replace_string = replace_string[:-1]
+        replace_string = replace_string.capitalize()
+        replace_string = '%s%s%s' % (' [', replace_string, '] ')
+        return replace_string
+
+    # Replace all <oppia-noninteractive-**> tags with their names captialized
+    # in square brackets.
+    html_content_string = re.sub(
+        r'<(oppia-noninteractive\s*?)[^>]+>', _replace_rte_tag,
```

Comment:
Couldn't you construct a regex that includes both the opening tag and the closing tag, rather than relying on this snippet subsequently being passed through html_cleaner? Conceptually you should be replacing the entirety of `<tag></tag>` with the descriptive name.

Reply:
Perhaps: ``'<(oppia-noninteractive\s*?)[^>]+></oppita-interactive.+?>'``

Reply:
I don't understand what you mean by "nested tags". Could you give an example?

Reply:
Is this resolved now given that you went with the suggested regex above?

Reply:
nested tags weren't actually a use case right now (or in the foreseeable future) so it's not an issue @BenHenning

Reply:
Ack, thanks for clarifying & that makes sense. In that case, this looks resolved.